wicket-jquery-ui/src/main/java/com/googlecode/wicket/jquery/ui/form/autocomplete/AutoCompleteTextField.java

```diff
@@ -323,9 +340,42 @@ protected CharSequence getChoiceCallbackUrl()
 
 				return super.$();
 			}
+
+			@Override
+			public void onConfigure(Component component) {
+				super.onConfigure(component);
+				if (preventSelectWhileQueryIsRunning) {
+					// we define special functions in for handling select and fetch data
+					this.setOption("select", "function(request, response) {\n" + getVarName() + ".select(request, response); \n}");
+					if (this.isEnabled(component))
+					{
+						this.setOption("source",  "function(event, ui) {\n" + getVarName() + ".fetchItems(event, ui); \n}");
+					}
+				}
+			}
 		};
 	}
 
+	public void renderHead(IHeaderResponse response) {
+		if (preventSelectWhileQueryIsRunning) {
+			response.render(JavaScriptHeaderItem.forReference(JS));
+			// we create a prototype with the data we need
+			response.render(OnDomReadyHeaderItem.forScript(getVarName() + " = new WJQUI.AutoComplete('"
+					+ getMarkupId() + "','" + choiceModelBehavior.getCallbackUrl()
+					+ "', '" + autoCompleteBehavior.getOnSelectAjaxBehavior().getCallbackUrl() + "');"));
+		}
+	}
+
+	private String getVarName() {
+		return "window.aut_" + getMarkupId();
```

Comment:
`aut_` is a bit vague. Maybe `autoCmpltTxtFld_` ?

Reply:
I will do this.